test/files/neg/nonunit-if.scala

```diff
@@ -0,0 +1,172 @@
+
+// scalac: -Werror -Wnonunit-statement -Wvalue-discard
+
+import collection.mutable.{LinkedHashSet, ListBuffer}
+import concurrent._
+
+// typo, misplaced expression during refactor
+class C {
+  import ExecutionContext.Implicits._
+  def c = {
+    def improved = Future(42)
+    def stale = Future(27)
+    improved              // warn
+    stale
+  }
+}
+// edited expression not deleted
+class D {
+  def d = {
+    class E
+    new E().toString      // warn
+    new E().toString * 2
+  }
+}
+// unused future in template
+class F {
+  import ExecutionContext.Implicits._
+  Future(42)              // warn
+}
+// unused template expression uses synthetic method of class
+case class K(s: String) {
+  copy()                  // warn
+}
+// mutations returning this are ok
+class Mutate {
+  val b = ListBuffer.empty[Int]
+  b += 42                 // nowarn, returns this.type
+  val xs = List(42)
+  27 +: xs                // warn
+
+  def f(x: Int): this.type = this
+  def g(): Unit = f(42)   // nowarn
+}
+// some uninteresting expressions may warn for other reasons
+class WhoCares {
+  null                    // warn for purity
+  ???                     // nowarn for impurity
+}
+// explicit Unit ascription to opt out of warning, even for funky applies
+class Absolution {
+  def f(i: Int): Int = i+1
+  import ExecutionContext.Implicits._
+  Future(42): Unit        // nowarn { F(42)(ctx) }: Unit where annot is on F(42), TODO spurious -Wvalue-discard
```

Comment:
This TODO looks dead.  But it looks like there are spurious warnings on the uni-branched if's below (foreach/forall), which would be good to not emit ideally.